scripts/kubectl-kubestellar-ensure-wmw

```diff
@@ -70,20 +72,57 @@ esac
 
 set -e
 
-original_ws=$(kubectl ws "${kubectl_flags[@]}" . --short)
+IFS=':' read -ra ws_array <<< "$wmw_name"
+first_item="true"
+parent=""
+kubectl config use-context $space_mgt
 
-if [ "$original_ws" != root ]; then
-    kubectl ws "${kubectl_flags[@]}" ..
-    if kubectl "${kubectl_flags[@]}" get workspaces.tenancy.kcp.io "$wmw_name" &> /dev/null; then
-	echo "Warning: parent has a child with the given WMW name; are you sure the right workspace is current?" >&2
-	sleep 15 # give a chance for ^C
-    fi
-    kubectl ws "${kubectl_flags[@]}" -
-fi
+create_space() {
+    kubectl "${kubectl_flags[@]}" apply -f - <<EOF
+apiVersion: space.kubestellar.io/v1alpha1
+kind: Space
+metadata:
+  name: $item
+  namespace: $spaceprovider_ns
+  annotations:
+    kubestellar.io/space-path: $parent
+spec:
+  SpaceProviderDescName: default
+  Managed: true
+EOF
+}
+
+for item in "${ws_array[@]}"; do
+    case "$item" in
+    (root)
+        if [ "$first_item" == "true" ]; then
+            parent=$item
+        else
+            echo "\"root\" workspace can only be used at the beginning"
+            exit 2
+        fi;;
+    (*)
+        if [ "$verbose" == "true" ]; then
+            if [[ -z "$(kubectl get space $item -n $spaceprovider_ns --ignore-not-found)" ]] ; then
+                echo "Space "$item" does not exist. Creating"
+                create_space
+                sleep 3
+            fi
+        else
+            if [[ -z "$(kubectl get space $item -n $spaceprovider_ns --ignore-not-found)" ]] ; then
+                create_space 1> /dev/null
+                sleep 3
+            fi
+        fi
+        parent=$parent":"$item;;
+    esac
+    first_item="false"
+done
 
-if kubectl "${kubectl_flags[@]}" get workspaces.tenancy.kcp.io "$wmw_name" &> /dev/null
-then kubectl ws "${kubectl_flags[@]}" "$wmw_name"
-else kubectl ws "${kubectl_flags[@]}" create "$wmw_name" --enter
+if ! kubectl ws "${kubectl_flags[@]}" "$wmw_name"
+then
+    echo "Can't access workspace ""$wmw_name"".Check space exist and ready"
```

Comment:
I doubt that the quotes here are doing what you intend.

```console
bash-3.2$ wmw_name=root:foo:baz

bash-3.2$     echo "Can't access workspace ""$wmw_name"".Check space exist and ready"
Can't access workspace root:foo:baz.Check space exist and ready
```

Do you know about `${wmw_name@Q}` ?  Did you know that MacOS ships with a version of bash so old that it does not support `@Q`?

Reply:
Will change